File: wlxx-framework/wlxx-apt/src/main/java/com/xzg/wlxx/apt/AutoFeign.java
package com.xzg.wlxx.apt;

import java.lang.annotation.*;

/**
 * @author XiaoZG
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface AutoFeign {

    String name() default "";
}

File: wlxx-framework/wlxx-apt/src/main/kotlin/com/xzg/wlxx/apt/AutoFeign.kt
package com.xzg.wlxx.apt

import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod

/**
 * @author XiaoZG
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(
    AnnotationRetention.RUNTIME
)
@MustBeDocumented
@RequestMapping(method = [RequestMethod.POST])
annotation class AutoFeign(val name: String = "")
